### PR TITLE
Fix linting errors & exclude intentional usage examples

### PR DESCRIPTION
**Pull request by int3l**
Since we have pre-commit config in place, I decided to run it against the repo -- got a lot of errors.
This PR fixes most errors and ignores some of the intentional examples (those are all `noqa` comments).
Let me know if you don't like any of the changes and sorry for the amount of changed files :D
I will run pre-commit on every commit from now on.

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -9,3 +9,4 @@ repos:
     rev: 3.8.3
     hooks:
     -   id: flake8
+        args: ['--ignore=E702,E722']
```
Comment (int3l) [inline .pre-commit-config.yaml:12]: Multiple statements on one line (semicolon) (E702)
Do not use bare except, specify exception instead (E722)
The usage of those two seems to be always intentional, this is why I ignore them globally.
Let me know if you want to ignore them in-place.

File: sample_code/ep025/rev03/flake8_named_arguments.py [hunk 2]
```diff
@@ -20,17 +20,17 @@ def __init__(self) -> None:
 
     def visit_Call(self, node: ast.Call) -> None:
         for keyword in node.keywords:
-            if (
-                    keyword.arg is None and
-                    isinstance(keyword.value, ast.Dict) and
+            if all(
```
Comment (int3l) [inline sample_code/ep025/rev03/flake8_named_arguments.py:23]: Using `all` as a solution for the W503/W504 warnings -- it is funny how both warnings are presented as solutions to each other via https://www.flake8rules.com/rules/W503.html and https://www.flake8rules.com/rules/W504.html :D

File: .pre-commit-config.yaml (hunk 1 above)
Comment (asottile) [reply .pre-commit-config.yaml:12]: `ignore` is almost always wrong -- it's what's triggering W503 and W504 at the same time, you want `extend-ignore` -- remove the arg here, set `noqa` on the individual lines, and don't change the `all(...)` it was correct how it was

but in reality, please do not ignore these, use noqa instead on the specific errors

**Review by asottile — COMMENTED** [review]
a lot of these change the code to be different from the video, I'm hesitant to change that and where possible would prefer what the video has

the whitespace changes and typo fixes are probably fine but maybe we should `exclude:` the code from flake8? idk

**Comment by int3l** [comment]
You are right, `noqa` is better.

File: sample_code/ep028/rev02/t.py [hunk 3]
```diff
@@ -5,7 +5,7 @@ def c_main(stdscr: 'curses._CursesWindow') -> int:
     while True:
         # rendering phase
         ...
-        char = stdscr.get_wch()
+        stdscr.get_wch()
```
Comment (int3l) [inline sample_code/ep028/rev02/t.py:8]: Do you want me to also revert the unused vars like this `char` assignment and the `args` for ArgParse?

Comment (asottile) [reply sample_code/ep028/rev02/t.py:8]: yeah, they were placeholders imaginging you'd use the arguments / characters afterwards

another idea might be to use `per-file-ignore`s to exclude the particular rules in particular files

Comment (int3l) [reply sample_code/ep028/rev02/t.py:8]: I saw that option, but which file should we use for the ignore metadata `setup.cfg`, `tox.ini` or `.flake8`? I am not sure.

Comment (asottile) [reply sample_code/ep028/rev02/t.py:8]: I usually prefer setup.cfg unless tox.ini exists

Comment (int3l) [reply sample_code/ep028/rev02/t.py]: Ok, we have the setup.cfg with flake8 `per-file-ignore`.

**Review by asottile — APPROVED** [review]

**Pull request merged**